### ISSUE TITLE
Pandoc job fails on release

### ISSUE THREAD

**Issue by jonringer** [NixOS/nixpkgs#97170]
ofborg and release channels are broken

example ofborg output: https://gist.github.com/GrahamcOfBorg/1f9a94055ab93cda2d381a535ed55493

did a git bisect find offending commit 05bf5320827

failure can be reproduced locally by running:
```
nix-build pkgs/top-level/release.nix -A unstable --show-trace
```
cc @peti

example output:
```
$ nix-build pkgs/top-level/release.nix -A unstable --show-trace
trace: `mkStrict' is obsolete; use `mkOverride 0' instead.
trace: `lib.nixpkgsVersion` is deprecated, use `lib.version` instead!
trace: Warning: `showVal` is deprecated and will be removed in the next release, please use `traceSeqN`
trace: lib.zip is deprecated, use lib.zipAttrsWith instead
error: while evaluating the attribute 'constituents' of the derivation 'nixpkgs-20.09pre1234.abcdef' at /home/jon/projects/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'pandoc.x86_64-linux' at undefined position:
attribute 'x86_64-linux' missing, at /home/jon/projects/nixpkgs/pkgs/top-level/release.nix:98:15
```

**Comment by cdepillabout** [comment]
This could be quickly fixed with a change like the following:

I'm testing on `master` on commit 32a898bb1, and this could be fixed by a quick change like the following:

```diff
diff --git a/pkgs/development/haskell-modules/hackage-packages.nix b/pkgs/development/haskell-modules/hackage-packages.nix
index 3c1f3ef5647..513ef0fc21c 100644
--- a/pkgs/development/haskell-modules/hackage-packages.nix
+++ b/pkgs/development/haskell-modules/hackage-packages.nix
@@ -189559,7 +189559,6 @@ self: {
        '';
        description = "Conversion between markup formats";
        license = stdenv.lib.licenses.gpl2Plus;
-       hydraPlatforms = stdenv.lib.platforms.none;
        maintainers = with stdenv.lib.maintainers; [ peti ];
      }) {};
 
```

This removes the `hydraPlatforms = none` line in `haskellPackages.pandoc_2_10_1`.

However, `hackage-packages.nix` is an auto-generated file, so it is possible this will be reverted next time the file is auto-generated.

I've seen the `hackage2nix` tool set `hydraPlatforms = none` for various reasons before, but I'm not sure what could be going on here.  Maybe @maralorn or @peti would know?

**Comment by worldofpeace** [comment]
I pushed https://github.com/NixOS/nixpkgs/commit/f30e80ba9ba2562a24310c5aed88ec1399a7d5e5. Even when `hackage-packages.nix` is auto-generated we don't cause any issues by editing it manually since it just gets overwritten. But we do need to make hackage2nix not set `hydraPlatforms` to none on things that are in `release.nix`.

**Comment by maralorn** [comment]
I think @peti will know immediately how to approach this correctly. This fix will do for a few days, then we will need a better solution in haskell-updates.
We should not forget this next friday.

**Comment by worldofpeace** [comment]
@maralorn My first guess is we need to do some post processing https://github.com/NixOS/cabal2nix/blob/107370041adfa96e891576852b5c4cc7e5bbd225/src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs#L153

**Comment by peti** [comment]
When Stackage LTS mandates use of a package version that's not the latest one, then we generate two copies of the package: the version mandated by LTS and the latest one -- but only the LTS version is actually built by Hydra. So when I updated Nixpkgs to use the latest Pandoc by default, we ended up having a package that's part of the `release.nix` jobset but which didn't list any platforms for Hydra builds.

The solution to this issue is to add a `doDistribute` function to the defnition of Pandoc to ensure that `meta.hydraPlatforms` is set.
